components/raftstore/src/store/read_queue.rs

```diff
@@ -230,14 +216,47 @@ where
         None
     }
 
-    pub fn advance_leader_reads<T>(&mut self, states: T)
+    pub fn advance_leader_reads<T>(&mut self, tag: &str, states: T)
     where
         T: IntoIterator<Item = (Uuid, Option<LockInfo>, u64)>,
     {
-        for (uuid, _, index) in states {
-            assert_eq!(uuid, self.reads[self.ready_cnt].id);
-            self.reads[self.ready_cnt].read_index = Some(index);
-            self.ready_cnt += 1;
+        let mut states_iter = states.into_iter();
+        while let Some((uuid, info, index)) = states_iter.next() {
+            let invalid_id = match self.reads.get_mut(self.ready_cnt) {
+                Some(r) if r.id == uuid => {
+                    r.read_index = Some(index);
+                    self.ready_cnt += 1;
+                    continue;
+                }
+                Some(r) => Some((r.id, r.propose_time)),
+                None => None,
+            };
+
+            if !self.contexts.contains_key(&uuid) {
+                debug!(
+                    "cannot find corresponding read from pending reads";
+                    "uuid" => ?uuid, "read-index" => index, "tag" => tag
+                );
+                continue;
+            }
+
+            error!("{} unexpected uuid detected", tag; "current_id" => ?invalid_id);
+            let mut expect_id_track = vec![];
+            for i in (0..self.ready_cnt).rev().take(10).rev() {
```

Comment:
Why is there `10` and `20`?

Reply:
It can be any number. `10` and `20` are just the safe number that I believe can find the overlap sequences.